src/FSharpLint.Core/Framework/ParseFile.fs

```diff
@@ -116,22 +89,43 @@ module ParseFile =
 
         { options with OtherOptions = otherOptions }
 
-    /// Parses a file using `FSharp.Compiler.Service`.
-    let parseFile file (checker:FSharpChecker) projectOptions =
-        let source = File.ReadAllText(file)
+    let private parse file source (checker:FSharpChecker) =
+        let sourceText = SourceText.ofString source
+        let projectOptions = getProjectOptionsFromScript checker file source
+        let parsingOptions = checker.GetParsingOptionsFromProjectOptions projectOptions |> fst
+        let parseResults =
+            checker.ParseFile(file, sourceText, parsingOptions)
+            |> Async.RunSynchronously
 
-        let projectOptions =
```

Comment:
The main thing I'm not sure about in this PR is that we're now always calling `getProjectOptionsFromScript` instead of possibly using the options which resulted from the call to Dotnet.ProjInfo. Do we lose anything as a result?

Reply:
You should be worried here. There are things like compiler defines that come from the project file that won't be read correctly in this case. Target framework profiles, etc as well.

I have to ask, what's the main motivation for dropping DPI? In my opinion, it's the most consistent and correct view of the project file you're going to get, because it actually has msbuild evaluate the project file and collect the F# compiler arguments that the SDK would use to compile that file.

Reply:
See my other reply for more info, but basically just trying to support multi-targeted projects. I'm convinced doing things this way is a bad idea now, so if we can figure out what the issue is with how we're using DPI and multi-targeted projects, then I'm happy to keep using it